src/Nest/QueryDsl/Geo/GeoCoordinateJsonConverter.cs

```diff
@@ -8,7 +8,7 @@ internal class GeoCoordinateJsonConverter : JsonConverter
 		public override bool CanConvert(Type objectType) => true;
 		public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
 		{
-			var p = value as GeoCoordinate;
+			var p = (GeoCoordinate)value;
```

Comment:
It would. yes 😄 

My thinking for converting to a direct cast however is that the converter is internal so we have control over which types it is applied to, so we should only be applying it to `GeoCoordinate`. If it's applied to another type other than `GeoCoordinate`, I think it would be better to have the invalid cast exception (to fix) rather than swallowing. In addition, the direct cast will slightly faster than `as`, which is effectively an `is` check and then cast.